File: mobile_app/lib/data/models/offer_model.dart
import '../../domain/entities/offer.dart';

class OfferModel extends Offer {
  OfferModel({
    required String id,
    required String title,
    required String description,
    required double discountPercentage,
    required double originalPrice,
    required double discountedPrice,
  }) : super(
          id: id,
          title: title,
          description: description,
          discountPercentage: discountPercentage,
          originalPrice: originalPrice,
          discountedPrice: discountedPrice,
        );

  factory OfferModel.fromJson(Map<String, dynamic> json) {
    return OfferModel(
      id: json['id'],
      title: json['title'],
      description: json['description'],
      discountPercentage: json['discountPercentage'].toDouble(),
      originalPrice: json['originalPrice'].toDouble(),
      discountedPrice: json['discountedPrice'].toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'description': description,
      'discountPercentage': discountPercentage,
      'originalPrice': originalPrice,
      'discountedPrice': discountedPrice,
    };
  }
}

File: backend/src/data/models/offer.model.ts
import { Schema, model } from 'mongoose';
import { Offer } from '../../domain/entities/offer.entity';

const offerSchema = new Schema<Offer>({
  title: { type: String, required: true },
  description: { type: String, required: true },
  discountPercentage: { type: Number, required: true },
  originalPrice: { type: Number, required: true },
  discountedPrice: { type: Number, required: true },
});

export const OfferModel = model<Offer>('Offer', offerSchema);